hs_app_netCDF/receivers.py

```diff
@@ -20,58 +21,61 @@
 from hs_file_types.models.netcdf import create_header_info_txt_file, add_metadata_to_list
 
 
-# receiver used to extract metadata after user click on "create resource"
-@receiver(pre_create_resource, sender=NetcdfResource)
-def netcdf_pre_create_resource(sender, **kwargs):
-    files = kwargs['files']
-    metadata = kwargs['metadata']
+@receiver(post_create_resource, sender=NetcdfResource)
+def netcdf_post_create_resource(sender, **kwargs):
+    log = logging.getLogger()
+    resource = kwargs['resource']
     validate_files_dict = kwargs['validate_files']
-    source_names = kwargs['source_names']
+    res_file = resource.files.all().first()
 
-    if __debug__:
-        assert(isinstance(source_names, list))
-
-    file_selected = False
-    in_file_name = ''
-    nc_file_name = ''
-    if files:
-        file_selected = True
-        in_file_name = files[0].file.name
-        nc_file_name = os.path.splitext(files[0].name)[0]
-    elif source_names:
-        nc_file_name = os.path.splitext(os.path.basename(source_names[0]))[0]
-        ref_tmpfiles = utils.get_fed_zone_files(source_names)
-        if ref_tmpfiles:
-            in_file_name = ref_tmpfiles[0]
-            file_selected = True
-
-    if file_selected and in_file_name:
-        # file validation and metadata extraction
-        nc_dataset = nc_utils.get_nc_dataset(in_file_name)
+    if res_file:
+        temp_file = utils.get_file_from_irods(res_file)
+        nc_dataset = nc_utils.get_nc_dataset(temp_file)
+        nc_file_name = res_file.file_name
 
         if isinstance(nc_dataset, netCDF4.Dataset):
             # Extract the metadata from netcdf file
-            res_dublin_core_meta, res_type_specific_meta = nc_meta.get_nc_meta_dict(in_file_name)
+            res_dublin_core_meta, res_type_specific_meta = nc_meta.get_nc_meta_dict(temp_file)
             # populate metadata list with extracted metadata
+            metadata = []
             add_metadata_to_list(metadata, res_dublin_core_meta, res_type_specific_meta)
+            for element in metadata:
+                # here k is the name of the element
+                # v is a dict of all element attributes/field names and field values
+                k, v = element.items()[0]
+                if k == 'title':
+                    # update title element
+                    title_element = resource.metadata.title
+                    resource.metadata.update_element('title', title_element.id, **v)
+                elif k == 'rights':
+                    rights_element = resource.metadata.rights
+                    resource.metadata.update_element('rights', rights_element.id, **v)
+                elif k == 'creator':
+                    new_creator = resource.metadata.create_element('creator', **v)
```

Comment:
@pkdash This is for the situation that when there is creator info in the netCDF file, the existing "creator" which is the resource owner will be deleted and make the creator info extracted from the netCDF file as the 1 order creator (or "author") in HydroShare.